app/services/msg/upcoming.rb

```diff
@@ -3,14 +3,14 @@ def respond
     events = Event.upcoming.limit(5)
     response = ""
     events.each do |event|
-      response += "Start #{event.start_time.to_s} " if event.start_time.present?
-      response += "End #{event.end_time.to_s} " if event.end_time.present?
+      response += "#{event.start_time.strftime('%a %b %d %k:%M')} - " if event.start_time.present?
+      response += "#{event.end_time.strftime('%b %d %k:%M')} " if event.end_time.present?
       response += " [#{event.id_code.to_s}] " if event.id_code.present?
-      response += event.title.truncate(15) if event.title.present?
+      response += event.title.truncate(23) if event.title.present?
```

Comment:
A better design might be to define this `response` string in a presenter method on an `Event`.  (whether it's an actual Decorator/Presenter class, an Event service object, or an Event instance method, all are reasonable.)  Then this entire service object could be replaced by, at most,

```ruby
   Event.upcoming.limit(5).map(&:response).join("\n")
```

And since this entire service object is about Events, the above line of code could be in an Event Presenter, an Event service object, or an Event class method.